### PR TITLE
Implement a pushRoute service to satisfy the FlutterView.pushRoute API

### LINKED ISSUE #10399: Implement pushRoute API on iOS
Hook up the `pushRoute` method on the `FlutterView` class on Android. Also something similar is desirable on iOS in order to be able to initiate Flutter code with different initial routes.

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/10399

### CODE REVIEW

File: packages/flutter/lib/src/widgets/binding.dart [hunk 1]
```diff
@@ -198,10 +206,20 @@ abstract class WidgetsBinding extends BindingBase implements GestureBinding, Ren
...
+      if (await observer.didPushRoute(route))
+        return;
+    }
+  }
+
   Future<dynamic> _handleNavigationInvocation(MethodCall methodCall) async {
     if (methodCall.method == 'popRoute')
       handlePopRoute();
-    // TODO(abarth): Handle 'pushRoute'.
+    if (methodCall.method == 'pushRoute')
```
Comment (mravn-google) [inline packages/flutter/lib/src/widgets/binding.dart]: `else if` or `switch`?

**Review by mravn-google — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
cc @abarth to see if this is what he had in mind

**Comment by Hixie** [comment]
this will need a test

File: packages/flutter/lib/src/widgets/binding.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter/lib/src/widgets/binding.dart]: changed to switch

**Comment by jason-simmons** [comment]
Added a test to widgets/binding_test

**Comment by szakarias** [comment]
This is great. It doesn't completely fix  #10399 since there is no support on iOS yet.

**Comment by abarth** [comment]
Looks reasonable.

**Pull request merged**